code-challenges.test.js

```diff
@@ -10,52 +10,169 @@
+//          something about replace
+const cryptex = string =>{
+    let stringThatWaz = string.replace(/a/gi, 4).replace(/e/gi, 3).replace(/i/gi, 1).replace(/o/gi, 0)
+    return stringThatWaz
+}
+
+
+//******if I have time******
+// function replacer(match, p1, p2, p3, offset, string) {
+//     // p1 is nondigits, p2 digits, and p3 non-alphanumerics
+//     return [p1, p2, p3].join(' - ');
+//   }
+//   let newString = 'abc12345#$*%'.replace(/([^\d]*)(\d*)([^\w]*)/, replacer);
+//   console.log(newString);  // abc - 12345 - #$*%
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
 
 
 
-// b) Create the function that makes the test pass.
 
 
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+// const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+// const letterA = "a"
+// // Expected output: ["Apple", "Banana", "Orange"]
+// const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+// const letterE = "e"
+// // Expected output: ["Cherry", "Blueberry", "Peach"]
 
 
 
 // b) Create the function that makes the test pass.
+describe("finder", () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    const letterA = "a"
+    let output1 = ["Apple", "Banana", "Orange"]
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    const letterE = "e"
+    let output2 = ["Cherry", "Blueberry", "Peach"]
+
+   it("takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+     expect(finder(arrayOfWords1, letterA)).toEqual(output1)
+     expect(finder(arrayOfWords2, letterE)).toEqual(output2)
+   })
+})
+// ReferenceError: finder is not defined
+//my function will take two arguments (arrayOfWords1, letterA)
+/// my function will need to itterate over the array of words
+//// looking for letterA    maybe .contain 
+///// might have to break up the array into strings.
+const finder = (array, letter) =>{
+    temp = []
+    array.filter(value=>{
+        if (value.includes(letter) || value.includes(letter.toUpperCase()))
+    //maybe the I can just check the value with conditonals
+        temp.push(value)
+    })
+    return temp
+}
+
```

Comment:
You logic looks great. I definitely think `.filter` is the correct choice. `.filter` will automatically return a new array AND it has a built in `if` statement. So you can take your exact logic and reduce the amount of code needed.
```
const finder = (array, letter) =>{
    return array.filter(value=>{
        return value.includes(letter) || value.includes(letter.toUpperCase())
    })
}
```